tests/components/samsungtv/test_config_flow.py

```diff
@@ -0,0 +1,388 @@
+        )
+        assert result["type"] == "abort"
+        assert result["reason"] == "not_supported"
+
+
+async def test_user_not_found(hass):
+    """Test starting a flow by user but no device found."""
+    with patch(
+        "homeassistant.components.samsungtv.config_flow.Remote",
+        side_effect=OSError("Boom"),
+    ), patch("homeassistant.components.samsungtv.config_flow.socket"):
+
+        # device not found
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": "user"}, data=MOCK_USER_DATA
+        )
+        assert result["type"] == "abort"
+        assert result["reason"] == "not_found"
+
+
+async def test_user_already_configured(hass, remote):
+    """Test starting a flow by user when already configured."""
+
+    # entry was added
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": "user"}, data=MOCK_USER_DATA
+    )
+    assert result["type"] == "create_entry"
+
+    # failed as already configured
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": "user"}, data=MOCK_USER_DATA
+    )
+    assert result["type"] == "abort"
+    assert result["reason"] == "already_configured"
+
+
+async def test_ssdp(hass, remote):
+    """Test starting a flow from discovery."""
+
+    # confirm to add the entry
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": "ssdp"}, data=MOCK_SSDP_DATA
+    )
+    assert result["type"] == "form"
+    assert result["step_id"] == "confirm"
+
+    # entry was added
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], user_input="whatever"
+    )
+    assert result["type"] == "create_entry"
+    assert result["title"] == "fake_name (fake_model)"
+    assert result["data"][CONF_HOST] == "fake_host"
+    assert result["data"][CONF_NAME] == "fake_name"
+    assert result["data"][CONF_MANUFACTURER] == "fake_manufacturer"
+    assert result["data"][CONF_MODEL] == "fake_model"
+    assert result["data"][CONF_ID] == "fake_uuid"
+
+
+async def test_ssdp_noprefix(hass, remote):
+    """Test starting a flow from discovery without prefixes."""
+
+    # confirm to add the entry
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": "ssdp"}, data=MOCK_SSDP_DATA_NOPREFIX
+    )
+    assert result["type"] == "form"
+    assert result["step_id"] == "confirm"
+
+    # entry was added
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], user_input="whatever"
+    )
+    assert result["type"] == "create_entry"
+    assert result["title"] == "fake2_name (fake2_model)"
+    assert result["data"][CONF_HOST] == "fake2_host"
+    assert result["data"][CONF_NAME] == "fake2_name"
+    assert result["data"][CONF_MANUFACTURER] == "fake2_manufacturer"
+    assert result["data"][CONF_MODEL] == "fake2_model"
+    assert result["data"][CONF_ID] == "fake2_uuid"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=4795320)